addons/stock/models/stock_location.py

```diff
@@ -71,6 +77,24 @@ def _compute_complete_name(self):
             else:
                 location.complete_name = location.name
 
+    @api.depends('cyclic_inventory_frequency', 'last_inventory_date', 'usage', 'company_id')
+    def _compute_next_inventory_date(self):
+        for location in self:
+            if location.company_id and location.usage in ['internal', 'transit'] and location.cyclic_inventory_frequency > 0:
```

Comment:
Shouldn't we allow view location also? If the user want an inventory on a part of the warehouse, a row, ...

Reply:
But we can't create an inventory for any locations except for when the usage is 'internal' and 'transit', right? Since `location_ids` has a `domain="[('company_id', '=', company_id), ('usage', 'in', ['internal', 'transit'])]")`.

Reply:
Okay. I saw a configuration where the user has a tree of views location and only the last node of the tree were internals location (the purpose was to avoid adding quantities in a non final loc). But anyway, if the feature doesn't work with classical inventory, we can skip it for this task.